gamerunner/user_code/get-user-code-from-github.js

```diff
@@ -103,17 +108,82 @@ function retrieveCode (users, category) {
+                    return updateDisableValues(user, error, true);
                 });
             } else {
                 /**
                  * @todo Detect 404 responses and have the ability to prune dead accounts after a certain
                  * number of failures.
                  */
-                console.warn('Unexpected response code:', response.statusCode, 'from', options.url, 'with message', body);
+                if (response.statusCode === 404) {
+                    console.warn('Unexpected response code:', response.statusCode, 'from', options.url, 'with message', body);
+                    return disableUser(user, `missing_${category}`);
+                }
             }
         });
     });
 }
 
+/**
+ * Using the updated user object, check to see if it has been update today, if not update user
+ * @param  {Number} id    - the users github_id
+ * @param  {String} error - The error for disable_message
+ */
+function disableUser ({ github_id }, error) {
+    return getUser(github_id)
+        .then(([user]) => !sameDay(user.last_update_date) ? updateDisableValues(user, error) : false )
+        .catch(e => console.warn(e));
+}
+
+/**
+ * Get the user from the db in case the last_update_date has been updated
+ * @param  {Number} id   - the users github_id
+ * @retrun {Object} user - the newly updated user object
+ * @retrun {Error}  err  - error
+ */
+function getUser (id) {
+  return new Promise((res, rej) => {
+      db.query(`SELECT * FROM player WHERE github_id = ${id}`, function (err, user) {
+          if (err) { return rej(err); }
+          return res(user);
+      });
+  });
+}
+
+/**
+ * Check to see if provided date is today
+ * @param  {Date} date - The last_update_date from the user object
+ * @return {Bool}      - True if same day, False if different days
+ */
+function sameDay (date) {
+    if (!date) return false;
+    const [update] = date.toISOString().split('T');
+    const [current] = new Date().toISOString().split('T');
+    return  (update === current);
+}
+
+/**
+ * Update the disable values, if called with success it removes all disabled based values
+ * @param  {Object} user    -  The updated user object
+ * @param  {String} error   - The error for disable_message
+ * @param  {Bool}   success - True if function is called after files have been writen
+ */
+function updateDisableValues (user, error, success) {
+    const date = success ? 'last_update_date = null' : 'last_update_date = localtimestamp';
+    const q = `UPDATE player SET disabled = $1, disabled_count = $2, disabled_message = $3, ${date} WHERE github_id = $4`;
+    const disabled = user.disabled_count >= 2 ? true : false;
+    const disabledCount = ++user.disabled_count;
+    const values = success ? [false, 0, null, user.githud_id] : [disabled, disabledCount, error, user.github_id];
+    db.update(q, values, (err, update) => {
+        count--;
+        if (err) {
+            throw err;
+        }
+        if (count === 0) {
```

Comment:
I think this needs to be before the error gets thrown. If an error occurred during the last update, I'm not sure the DB connection would close as-is.